src/components/Plans/PlanCalculator/PlanCalculatorButtons/index.js

```diff
@@ -0,0 +1,60 @@
+import React from 'react';
+import { useIntl } from 'react-intl';
+import { useQuery } from '../../../../hooks/useQuery';
+import { InjectAppServices } from '../../../../services/pure-di';
+import { TooltipContainer } from '../../../TooltipContainer/TooltipContainer';
+import * as S from './index.styles';
+
+export const PlanCalculatorButtons = InjectAppServices(
+  ({ selectedPlanId, selectedDiscountId, dependencies: { appSessionRef } }) => {
+    const intl = useIntl();
+    const _ = (id, values) => intl.formatMessage({ id: id }, values);
+    const query = useQuery();
+    const promoCode = query.get('promo-code');
+
+    const sessionPlan = appSessionRef.current.userData.user;
+    const isEqualPlan = sessionPlan.plan.idPlan === selectedPlanId;
+
+    return (
+      <div className="dp-container">
+        <div className="dp-rowflex">
+          <div className="dp-align-center dp-cta-plans">
+            <button
+              type="button"
+              className="dp-button button-medium primary-grey"
+              onClick={() => window.history.back()}
+            >
+              {_('plan_calculator.button_back')}
+            </button>
+            <TooltipContainer
+              visible={isEqualPlan}
+              content={_('plan_calculator.button_purchase_tooltip')}
+              orientation="top"
+            >
+              <S.PurchaseLink
+                className={`dp-button button-medium primary-green ${isEqualPlan ? 'disabled' : ''}`}
+                href={getBuyPurchaseUrl({
+                  controlPanelUrl: _('common.control_panel_section_url'),
+                  planId: selectedPlanId,
+                  discountId: selectedDiscountId,
+                  promoCode,
+                })}
+              >
+                {_('plan_calculator.button_purchase')}
+              </S.PurchaseLink>
+            </TooltipContainer>
+          </div>
+        </div>
+      </div>
+    );
+  },
+);
+
+const getBuyPurchaseUrl = ({ controlPanelUrl, planId, discountId, promoCode }) => {
+  return (
+    controlPanelUrl +
+    `/AccountPreferences/UpgradeAccountStep2?IdUserTypePlan=${planId}&fromStep1=True` +
```

Comment:
in this case it's no necessary to close literals line by line and use '+' , directly you can do it this way : 
```
 `/AccountPreferences/UpgradeAccountStep2?IdUserTypePlan=${planId}&fromStep1=True
       ${discountId ? `&IdDiscountPlan=${discountId}` : ''}
       ${promoCode ? `&PromoCode=${promoCode}` : ''}`
```

Reply:
:+1:

Reply:
> in this case it's no necessary to close literals line by line and use '+' , directly you can do it this way :

But in that case, will it not include the line-endings and spaces?

Reply:
would this be okay?

![image](https://user-images.githubusercontent.com/84402180/139925320-653f1b38-6eea-46ce-8796-efdae9f21c49.png)

Reply:
🤷‍♂️

Reply:
yes, itk Ok, maybe could be better this way:  ( each js code expresion in one line)

```
 `/AccountPreferences/UpgradeAccountStep2?IdUserTypePlan=${planId}&fromStep1=True
       ${discountId ? `&IdDiscountPlan=${discountId}` : ''}
       ${promoCode ? `&PromoCode=${promoCode}` : ''}`
```

Reply:
> > in this case it's no necessary to close literals line by line and use '+' , directly you can do it this way :
> 
> But in that case, will it not include the line-endings and spaces?

it will not include the line-endings, in case of spaces, you have to be sure not to have any space in the end of each line.

Reply:
![image](https://user-images.githubusercontent.com/1157864/139935641-dcc29dfd-a1ce-4e1c-9c1a-2719522c5764.png)

Reply:
Tal vez lo mejor sea no manejar las URL con concatenación de strings, eso es peligroso, y se nos pueden escapar los encodings.

Por ahí en este caso no es necesario, pero lo mejor sería usar una librería de templates de URLs, algo así:

```
var urlTemplate = '/AccountPreferences/UpgradeAccountStep2{?IdUserTypePlan}&fromStep1=True{?IdDiscountPlan}{?promoCode}'
var urlData = { IdUserTypePlan: planId, IdDiscountPlan: discountId, PromoCode: promoCode };
var url = hydrateTemplate(urlTemplate, urlData);
```

Reply:
Interesting discussion: https://stackoverflow.com/questions/10464152/uri-templates-is-there-an-rfc-6570-implementation-in-javascript

I will read it.